lib/front_desk.rb

```diff
@@ -0,0 +1,116 @@
+require_relative "rooms.rb"
+require_relative "reservations.rb"
+
+
+class Front_Desk
+
+  attr_reader :reservations, :rooms, :block_hold
+
+  def initialize
+    @reservations = []
+    @block_reservations = []
+    @rooms = load_rooms
+  end
+
+  def load_rooms
+    room_number = 0
+    rooms = []
+
+    20.times do
+      room_number += 1
+      rooms << Room.new({
+        :room_number => room_number
+      })
+    end
+    return rooms
+  end
+
+  def reserve_room(room_number,start_date, end_date)
+
+    available_rooms = available_rooms(start_date,end_date)
+    if !available_rooms.find { |room| room.room_number == room_number }
+      raise StandardError
```

Comment:
Instead of raising a `StandardError`, best practice is to define your own appropriately named exception type that inherits from `StandardError` and raise that. The reason is that `StandardError` is very broad, and if the code that calls this method has to `rescue StandardError`, that will catch things like `ArgumentError`s and `NameError`s that indicate true bugs.